pkg/trace/agent/obfuscate.go

```diff
@@ -122,3 +126,23 @@ func (a *Agent) obfuscateStatsGroup(b *pb.ClientGroupedStats) {
 		b.Resource = o.QuantizeRedisString(b.Resource)
 	}
 }
+
+var (
+	obfuscatorLock sync.Mutex
+)
+
+func (a *Agent) lazyInitObfuscator() *obfuscate.Obfuscator {
+	// Ensure thread safe initialization
+	obfuscatorLock.Lock()
```

Comment:
Would a `sync.Once` work a bit better here maybe? Or do we have uses where we want to set the obfuscator to nil and re-initialize it (so that this logic happens more than once)?

Reply:
`sync.Once` causes problems with tests, unfortunately. It's not ideal to shape code around tests but the alternative is a major refactor of many tests.